wlm/src/main/java/ai/djl/serving/wlm/ModelInfo.java

```diff
@@ -648,11 +658,19 @@ void checkAvailableMemory(Device device) throws IOException {
         if (requiredMemory <= 0 && tpDegree < 1) {
             // TODO: handle LMI use case in future
             try (Stream<Path> walk = Files.walk(modelDir)) {
-                // estimate the memory to be 1.2x of file size
                 requiredMemory =
                         walk.filter(Files::isRegularFile).mapToLong(ModelInfo::getFileSize).sum();
-                requiredMemory = requiredMemory * 12 / 10;
             }
+            if (downloadS3Dir != null) {
+                try (Stream<Path> walk = Files.walk(downloadS3Dir)) {
+                    requiredMemory +=
+                            walk.filter(Files::isRegularFile)
+                                    .mapToLong(ModelInfo::getFileSize)
+                                    .sum();
+                }
+            }
+            // estimate the memory to be 1.2x of file size
+            requiredMemory = requiredMemory * 12 / 10;
```

Comment:
this memory estimation should only apply on CPU, not GPU. Or there could be some issues. For now, let's not apply a factor of 1.2, just follow it's original size. 

Apart from that, you will never know how user plan to to load these files. For example, if customer do `load_with_low_cpu_memory` option on HuggingFace. The peak model memory will remains at per checkpoint size. So I would suggest to now estimate by this.

Reply:
The current estimation doesn't apply to LMI use case.

For a real production usage, user should provide a more accurate memory value in the model. I think it's better to be more conservative if user didn't provide the memory footprint.

Reply:
This will break many default use cases. All of our notebook has no preset estimated memory. There will be definitely bumpy for many customers to try especially CPU space is not enough.

For example, our AOT will partition model to 46GB 4 shards. If user didn't set memory, it will break here.

Reply:
Although for AWS there is no case CPU memory is less than GPU memory, but many desktop customer may have 8GB CPU memory with 3060 (16GB) GPU card

Reply:
DeepSpeed, HuggingFace, FasterTransformer has mechanism to load checkpoint by checkpoint

Reply:
the estimated memory will be 0 for LMI case (tp > 1). This won't break our notebook.

Reply:
What if the S3 downloaded data is a lookup table and only used when needed? What if not all data downloaded by S3 is used all in the same time. Like parquet file, why should I worry the total size?

Reply:
It will break many default "work" use cases through forcibly setting a CPU memory to bypass this line